sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

```diff
@@ -104,6 +111,98 @@ public IScaleMonitor GetMonitor()
             return _scaleMonitor.Value;
         }
 
+        internal class EventProcessorStoredEventsManager : IDisposable
+        {
+            private SemaphoreSlim _storedEventsGuard;
+            private int _maxBatchSize;
+            private int _minBatchSize;
+
+            // This is internal for mocking purposes only.
+            internal ConcurrentDictionary<string, List<EventData>> StoredEvents { get; set; }
+
+            public bool HasStoredEvents
+            {
+                get
+                {
+                    try
+                    {
+                        _storedEventsGuard.Wait();
```

Comment:
I don't believe that we need to take the semaphore here.   The concurrent dictionary is thread-safe.